btx/processing/pipca.py

```diff
@@ -0,0 +1,1011 @@
+            norm = True
+            norm_str = "Normalized " if norm else ""
+
+            ipca_loss = self.compression_loss(X, U[:, :q_pca], normalized=norm)
+            print(f"iPCA {norm_str}Compression Loss: {ipca_loss}")
+
+            pca_loss = self.compression_loss(X, U_pca[:, :q_pca], normalized=norm)
+            print(f"PCA {norm_str}Compression Loss: {pca_loss}")
+
+            print("\n")
+            ipca_tot_var = np.sum(var)
+            pca_tot_var = np.sum(var_pca)
+
+            print(f"iPCA Total Variance: {ipca_tot_var}")
+            print(f"PCA Total Variance: {pca_tot_var}")
+            print("\n")
+
+            ipca_exp_var = (np.sum(S[:q_pca] ** 2) / (x - 1)) / np.sum(var)
+            print(f"iPCA Explained Variance: {ipca_exp_var}")
+
+            pca_exp_var = (np.sum(S_pca[:q_pca] ** 2) / (x - 1)) / np.sum(var_pca)
+            print(f"PCA Explained Variance: {pca_exp_var}")
+            print("\n")
+
+            print("iPCA Singular Values: \n")
+            print(S)
+            print("\n")
+
+            print("PCA Singular Values: \n")
+            print(S_pca[:q_pca])
+            print("\n")
+
+            mean_inner_prod = np.inner(mu.flatten(), mu_pca.flatten()) / (
+                np.linalg.norm(mu) * np.linalg.norm(mu_pca)
+            )
+
+            print("Normalized Mean Inner Product: " + f"{mean_inner_prod}")
+            print("\n")
+
+            print("Basis Inner Product: \n")
+            print(np.diagonal(np.abs(U[:, :q_pca].T @ U_pca[:, :q_pca])))
+
+            ipca_mu_u = np.hstack((mu / np.linalg.norm(mu), U[:, :q_pca]))
+            pca_mu_u = np.hstack((mu_pca / np.linalg.norm(mu_pca), U_pca[:, :q_pca]))
+
+            b = plt.imshow(np.abs(ipca_mu_u.T @ pca_mu_u))
+            plt.colorbar(b)
+            plt.savefig(f"fig_{q}_{self.size}.png")
+            plt.show()
+
+            print("\n")
+            self.report_interval_data()
+
+        finally:
+            # reset counter
+            self.psi.counter = event_index
+
+
+    def report_interval_data(self):
+        """
+        Method to print out iPCA time interval data.
+        """
+
+        if self.rank != 0:
+            return
+
+        task_durations = self.task_durations
+
+        if not len(task_durations):
+            print("No model updates or initialization recorded.")
+            return
+
+        # log data
+        print("\n")
+        for key in list(task_durations.keys()):
+            interval_mean = np.mean(task_durations[key])
+            print(
+                "Mean per-batch compute time of step "
+                + f"'{key}': {interval_mean:.4g}s"
+            )
+
+    def save_interval_data(self):
+        """
+        Save time interval data gathered during iPCA.
+
+        Parameters
+        ----------
+        save_data : bool, optional
+            if True, save interval data to file in self.output_dir, by default False
```

Comment:
Currently not a parameter. Should it be made one?

Reply:
Hi @apeck12 , I think at one point this file had additional functionality that was spun off. Now that this function's explicit purpose is to save the interval data I'll remove this extraneous parameter.